src/languages/smtlib/parseSmtlib.mly

```diff
@@ -114,11 +128,30 @@ sorted_var:
       let loc = L.mk_pos $startpos $endpos in T.colon ~loc c ty }
 ;
 
+constructor_dec:
+  | OPEN s=SYMBOL args=sorted_var* CLOSE
+    { let c = I.(mk term s) in c, args }
+;
+
+datatype_dec:
+  | OPEN cl=constructor_dec+ CLOSE
+    { [], cl }
+  | OPEN PAR OPEN args=SYMBOL+ CLOSE OPEN cl=constructor_dec+ CLOSE
+    { let pl = List.map I.(mk sort) args in pl, cl }
+;
+
+fun_dec:
+  | OPEN s=SYMBOL OPEN args=sorted_var* CLOSE ty=sort CLOSE
+    { let f = I.(mk term s) in
+      f, (args, ty) }
+
 term:
   | c=spec_constant
     { c I.term }
   | s=qual_identifier
     { let loc = L.mk_pos $startpos $endpos in T.apply ~loc s [] }
+  | OPEN NOT t=term CLOSE
+    { let loc = L.mk_pos $startpos $endpos in T.apply ~loc (T.const ~loc I.(mk term "not")) [t] }
```

Comment:
Same as in the lexer, why do you treat `not` specially ?

Reply:
Fine for me.